src/Spaces.ts

```diff
@@ -1,15 +1,32 @@
-import { Types } from 'ably';
+import { Types, Realtime } from 'ably';
 import SpaceOptions from './options/SpaceOptions';
 import Space from './Space';
-
 class Spaces {
   private spaces: Record<string, Space>;
   private channel: Types.RealtimeChannelPromise;
   ably: Types.RealtimePromise;
 
-  constructor(ably: Types.RealtimePromise) {
+  readonly version = '0.0.1';
+
+  constructor(optionsOrAbly: Types.RealtimePromise | Types.ClientOptions | string) {
     this.spaces = {};
-    this.ably = ably;
+    if (optionsOrAbly instanceof Realtime) {
+      this.ably = optionsOrAbly as Types.RealtimePromise;
+      this.addAgent(this.ably['options']);
+    } else {
+      let options: Types.ClientOptions = typeof optionsOrAbly === 'string' ? { key: optionsOrAbly } : optionsOrAbly;
+      this.addAgent(options);
+      this.ably = new Realtime.Promise(options);
+    }
```

Comment:
This changes significantly the signature of the `Spaces` constructor. I don't think we should be making changes like this in PRs, without discussion and updating relevant documentation first, unless it fixes buggy behaviour.